sensing/autoware_pointcloud_preprocessor/src/concatenate_data/concatenate_pointclouds.cpp

```diff
@@ -446,12 +503,35 @@ void PointCloudConcatenationComponent::cloud_callback(
   }
 }
 
+void PointCloudConcatenationComponent::single_cloud_callback(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr & input_ptr, const std::string & topic_name)
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  sensor_msgs::msg::PointCloud2::SharedPtr xyzirc_input_ptr(new sensor_msgs::msg::PointCloud2());
+  auto input = std::make_shared<sensor_msgs::msg::PointCloud2>(*input_ptr);
+  if (input->data.empty()) {
+    RCLCPP_WARN_THROTTLE(this->get_logger(), *this->get_clock(), 1000, "Empty sensor points!");
+  } else {
+    // convert to XYZI pointcloud if pointcloud is not empty
+    convertToXYZIRCCloud(input, xyzirc_input_ptr);
+  }
+
+  cloud_stdmap_[topic_name] = xyzirc_input_ptr;
+
+  timer_->cancel();
+  publishSingleLidar();
+}
+
 void PointCloudConcatenationComponent::timer_callback()
 {
   using std::chrono_literals::operator""ms;
   timer_->cancel();
   if (mutex_.try_lock()) {
-    publish();
+    if (input_topics_.size() == 1) {
+      publishSingleLidar();
+    } else {
+      publish();
```

Comment:
Perhaps, this ```publish()``` is triggered even when ```input_topics_.size() == 0```. Let's ensure non-zero case by:
```
} else if (input_topics_.size() > 1) {
```

not:

```
} else {
```

:+1: